test_app/tests/authentication/utils/test_claims.py

```diff
@@ -5,88 +5,233 @@
 @pytest.mark.parametrize(
-    "triggers, map_type, attrs, groups, exp_access_allowed, exp_is_superuser, exp_is_system_auditor, exp_claims, exp_last_login_map_results",
+    "triggers, map_type, role, attrs, groups, exp_access_allowed, exp_is_superuser, exp_is_system_auditor, exp_claims, exp_last_login_map_results",
     [
-        ({"always": {}}, "is_superuser", {}, [], True, True, None, {"team_membership": {}, "organization_membership": {}}, [{1: True}]),
-        ({"never": {}}, "is_superuser", {}, [], True, False, None, {"team_membership": {}, "organization_membership": {}}, [{1: False}]),
-        ({"always": {}}, "is_system_auditor", {}, [], True, None, True, {"team_membership": {}, "organization_membership": {}}, [{1: True}]),
-        ({"badkey": {}}, "is_system_auditor", {}, [], True, None, None, {"team_membership": {}, "organization_membership": {}}, [{1: "invalid"}]),
-        ({}, "is_system_auditor", {}, [], True, None, None, {"team_membership": {}, "organization_membership": {}}, [{1: "skipped"}]),
         (
-            {"always": {}, "never": {}},
+            {"always": {}},
             "is_superuser",
+            None,
+            {},
+            [],
+            True,
+            True,
+            None,
+            {"team_membership": {}, "organization_membership": {}, 'rbac_roles': {'system': {'roles': {}}, 'organizations': {}}},
+            [{1: True}],
+        ),
+        (
+            {"never": {}},
+            "is_superuser",
+            None,
             {},
             [],
             True,
             False,
             None,
-            {"team_membership": {}, "organization_membership": {}},
+            {"team_membership": {}, "organization_membership": {}, 'rbac_roles': {'system': {'roles': {}}, 'organizations': {}}},
+            [{1: False}],
+        ),
+        (
+            {"always": {}},
+            "is_system_auditor",
+            None,
+            {},
+            [],
+            True,
+            None,
+            True,
+            {"team_membership": {}, "organization_membership": {}, 'rbac_roles': {'system': {'roles': {}}, 'organizations': {}}},
+            [{1: True}],
+        ),
+        (
+            {"badkey": {}},
+            "is_system_auditor",
+            None,
+            {},
+            [],
+            True,
+            None,
+            None,
+            {"team_membership": {}, "organization_membership": {}, 'rbac_roles': {'system': {'roles': {}}, 'organizations': {}}},
+            [{1: "invalid"}],
+        ),
+        (
+            {},
+            "is_system_auditor",
+            None,
+            {},
+            [],
+            True,
+            None,
+            None,
+            {"team_membership": {}, "organization_membership": {}, 'rbac_roles': {'system': {'roles': {}}, 'organizations': {}}},
+            [{1: "skipped"}],
+        ),
+        # This one is ambiguous, create_claims() evaluates triggers in wrong order
```

Comment:
Is this a bug or am I misunderstanding this comment? If its not a bug can we elaborate on the comment and delete the test? If it is a bug can we add something like "Once xyz is reaolved uncomment this test".

Reply:
I was quite surprised, that the order of `always and never` wasn't important, it was always processed as `never`. So I'm not quite sure it is a bug or not, I'll try to look at it

Reply:
-> test removed

Reply:
That's your test, John :) I just removed the `xfail` and extended the params in `pytest.parametrize`

Reply:
anyway we are in agreement that the test is not useful, so this is solved